Recipes/InheritedTypeJsonConverter/InheritedTypeJsonConverter.cs

```diff
@@ -239,5 +194,69 @@ private IEnumerable<Type> GetChildTypes(Type type)
 
             return allChildTypes[type];
         }
+
+        private IEnumerable<CandidateChildType> GetCandidateChildTypes(Type objectType, HashSet<string> jsonProperties)
+        {
+            var childTypes = GetChildTypes(objectType);
```

Comment:
No big deal, but I think you get a slight flexibility gain in your code by moving the GetChildTypes to the calling method and not encapsulating this. Should another bit of code ever have need of a method that takes an IEnumerable<Type> to get Candidates from, we're one step closer. Plus the code now starts to become chainable. GetChildTypes().GetCandidates().SelectBestMatch()...